rosidl_cmake/cmake/rosidl_generate_action_interfaces.cmake

```diff
@@ -0,0 +1,81 @@
+# Copyright 2018 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Generate Interfaces for Actions
+#
+# This executes the extension point ``rosidl_generate_action_interfaces``.
+# An extension of this type should expect a list of `.action` files and
+# generate new files in response.
+# Extensions should create a target that generates the files with the input
+# `.action` file as a dependency.
+#
+# :param target: the _name of the generation target,
+#   specific generators might use the _name as a prefix for their own
+#   generation step
+# :type target: string
+# :param ARGN: a list of include directories where each value might
+#   be either an absolute path or path relative to the
+#   CMAKE_INSTALL_PREFIX.
+# :type ARGN: list of strings
+# :param TARGET_DEPENDENCIES: cmake targets or files the generated target
+#   should depend on
+# :type TARGET_DEPENDENCIES: list of strings
+# :param LIBRARY_NAME: the base name of the library, specific generators might
+#   append their own suffix
+# :type LIBRARY_NAME: string
+# :param SKIP_INSTALL: if set skip installing the interface files
+# :type SKIP_INSTALL: option
+# :param ADD_LINTER_TESTS: if set lint the interface files using
+#   the ``ament_lint`` package
+# :type ADD_LINTER_TESTS: option
+#
+# @public
+#
+function(rosidl_generate_action_interfaces target)
+  cmake_parse_arguments(_ARG
+    "ADD_LINTER_TESTS;SKIP_INSTALL"
+    "LIBRARY_NAME" "TARGET_DEPENDENCIES"
+    ${ARGN})
+  if(NOT _ARG_UNPARSED_ARGUMENTS)
+    message(FATAL_ERROR "rosidl_generate_action_interfaces() called without any idl "
+      "files")
+  endif()
+
+  set(_idl_files ${_ARG_UNPARSED_ARGUMENTS})
+
+  # Create a custom target
+  set(_sub_target "${target}+convert_action_to_msg_and_srv")
```

Comment:
@sloretz is that '+' a typo?

Reply:
The `+` is intentional. I figured it would make it more unlikely to conflict with another target.

Reply:
I see, should this one match with https://github.com/ros2/rosidl/pull/310/files#diff-bb53b2726c88913189fccc2d713745f9R138?

Reply:
Actually, this is strange.  I thought I deleted this file. The target should only be defined once in `rosidl_generate_interfaces` and passed to the cmake function that does the conversion